### ISSUE TITLE
Dash 1.0.0 data_table 4.0.0   Bad object: { "row_index": "odd" }

### ISSUE THREAD

**Issue by wvwhome** [plotly/dash-table#482]
Thanks for your interest in Plotly's Dash DataTable component!!

Note that GitHub issues in this repo are reserved for bug reports and feature
requests. Implementation questions should be discussed in our
[Dash Community Forum](https://community.plot.ly/c/dash).

Before opening a new issue, please search through existing issues (including
closed issues) and the [Dash Community Forum](https://community.plot.ly/c/dash).
If your problem or idea has not been addressed yet, feel free to
[open an issue](https://github.com/plotly/plotly.py/issues/new).

When reporting a bug, please include a reproducible example! We recommend using
the [latest version](https://github.com/plotly/dash-table/blob/master/CHANGELOG.md)
as this project is frequently updated. Issues can be browser-specific so
it's usually helpful to mention the browser and version that you are using.

Thanks for taking the time to help up improve this component!

**Comment by wvwhome** [comment]
Hi,
I just upgraded to dash 1.0.0 and dash_table 4.0.0   mostly good.  Thanks.
This example from  https://dash.plot.ly/datatable/style fails on both Windows 10 and Cent OS 7

The program with embedded error as comments follows:

#  06/25/2019
#  from  https://dash.plot.ly/datatable/style   Striped Rows
#
#    Error:
#
# Failed component prop type: Invalid component prop
# `style_cell_conditional[0].if` key `row_index` supplied to DataTable.
# Bad object: { "row_index": "odd" }
# Valid keys: [ "column_id", "column_type" ]
#
import dash
import dash_core_components as dcc
import dash_html_components as html
import dash_table
from collections import OrderedDict
import pandas as pd
import platform

print()
print('*** program starting ***')
print('Python:    ', platform.python_version())
print('dash:      ', dash.__version__)
print('dcc:       ', dcc.__version__)
print('html:      ', html.__version__)
print('dash_table:', dash_table.__version__)


# *** program starting ***
# Python:     3.7.3
# dash:       1.0.0
# dcc:        1.0.0
# html:       1.0.0
# dash_table: 4.0.0
# Running on http://127.0.0.1:8050/

external_stylesheets = ['https://codepen.io/chriddyp/pen/bWLwgP.css']

app = dash.Dash(__name__, external_stylesheets=external_stylesheets)

data = OrderedDict(
    [
        ("Date", ["2015-01-01", "2015-10-24", "2016-05-10", "2017-01-10", "2018-05-10", "2018-08-15"]),
        ("Region", ["Montreal", "Toronto", "New York City", "Miami", "San Francisco", "London"]),
        ("Temperature", [1, -20, 3.512, 4, 10423, -441.2]),
        ("Humidity", [10, 20, 30, 40, 50, 60]),
        ("Pressure", [2, 10924, 3912, -10, 3591.2, 15]),
    ]
)

df = pd.DataFrame(data)


app.layout = html.Div([

        dash_table.DataTable(
            data=df.to_dict('records'),
            columns=[{'id': c, 'name': c} for c in df.columns],

            style_cell_conditional=[
                {
                    'if': {'row_index': 'odd'},
                    'backgroundColor': 'rgb(248, 248, 248)'
                }
            ] + [
                {
                    'if': {'column_id': c},
                    'textAlign': 'left'
                } for c in ['Date', 'Region']
            ],
            style_header={
                'backgroundColor': 'white',
                'fontWeight': 'bold'
            }
        )
    ])


if __name__ == '__main__':
    app.run_server(debug=True)

**Comment by wvwhome** [comment]
Hmmm ... markdown was applied to the program code ... my oversight.

**Comment by alexcjohnson** [comment]
Ah, that's a mistake in fact - it doesn't make sense to talk about `row_index` for `style_cell_conditional` since that applies to headers as well as data. `row_index` can only be used in `style_data_conditional`. Thanks for pointing this out, we'll clean that up in the docs.

**Comment by alexcjohnson** [comment]
Digging in further: `style_cell_conditional` does support `row_index`, as long as you disable prop checks (which is the case by default in production mode, such as on our public docs site). But it does a weird thing: it separately counts up rows in headers and data cells. So if you only have one header, it's even (index 0) and so is the first data row.

Anyway, I'm going to fix the usage in the docs, and consider it a bug here that `row_index` is not ignored in `style_cell_conditional` if it isn't blocked by the prop check.

Thanks again for pointing this out @wvwhome!

**Comment by wvwhome** [comment]
Thanks for the further explanation, since the row_index was working for me when I loaded the program into the Google Cloud Platform -- now this makes more sense,

**Comment by chnaltn** [comment]
> Digging in further: `style_cell_conditional` does support `row_index`, as long as you disable prop checks (which is the case by default in production mode, such as on our public docs site). But it does a weird thing: it separately counts up rows in headers and data cells. So if you only have one header, it's even (index 0) and so is the first data row.
> 
> Anyway, I'm going to fix the usage in the docs, and consider it a bug here that `row_index` is not ignored in `style_cell_conditional` if it isn't blocked by the prop check.
> 
> Thanks again for pointing this out @wvwhome!

Hi,

So is this issue fixed? Because in documentation I see the same example. But it is not working on my app. I see the same "bad object" error. But it is working when I turn off the debug mode.

**Comment by Marc-Andre-Rivet** [comment]
Closing. Documentation was updated with https://github.com/plotly/dash-docs/pull/594 and there are no documentation examples using `style_cell_conditional.$index.if.row_index` at this time.

Table props definition does not define `row_index` for `style_cell_conditional` and allows `row_index` for `style_data_conditional` and `header_index` for `style_header_conditional`
